commontype/fee_config.go

```diff
@@ -144,6 +146,65 @@ func (f *FeeConfig) checkByteLens() error {
 	return nil
 }
 
+func (c *FeeConfig) getBigIntToSerialize() []**big.Int {
+	return []**big.Int{
+		&c.GasLimit, &c.MinBaseFee, &c.TargetGas, &c.BaseFeeChangeDenominator,
+		&c.MinBlockGasCost, &c.MaxBlockGasCost, &c.BlockGasCostStep,
+	}
+}
+
+func (c *FeeConfig) MarshalBinary() ([]byte, error) {
+	p := wrappers.Packer{
+		Bytes:   []byte{},
+		MaxSize: 1 * units.MiB,
+	}
+
+	for _, bigint := range c.getBigIntToSerialize() {
+		p.PackBool(*bigint == nil)
+		if p.Err != nil {
+			return nil, p.Err
+		}
+		if bigint != nil {
+			p.PackBytes((*bigint).Bytes())
+			if p.Err != nil {
+				return nil, p.Err
+			}
+		}
+	}
+	p.PackLong(c.TargetBlockRate)
+	if p.Err != nil {
+		return nil, p.Err
+	}
+
+	return p.Bytes, nil
+}
+
+func (c *FeeConfig) UnmarshalBinary(data []byte) error {
+	p := wrappers.Packer{
+		Bytes: data,
+	}
+	for _, bigint := range c.getBigIntToSerialize() {
+		isNil := p.UnpackBool()
+		if p.Err != nil {
+			return p.Err
+		}
+		if isNil {
+			continue
+		}
+		*bigint = big.NewInt(0).SetBytes(p.UnpackBytes())
```

Comment:
Should these be fixed bytes of size 32 since that is the max size allowed for these variables?

Reply:
[It is variable size](https://pkg.go.dev/math/big#Int.Bytes)

Reply:
Fixed it in e8ffb133